### PR TITLE
Resolve JS-81- Options Schemas required for ESLint v9

### PR DESCRIPTION
**Pull request by ericmorand-sonarsource**
Just testing from the CI to prevent my computer from crashing.

Fixes JS-81

### CODE REVIEW

File: sonar-plugin/javascript-checks/src/test/java/org/sonar/javascript/checks/NoNestedFunctionsCheckTest.java
Comment (ericmorand-sonarsource) [inline sonar-plugin/javascript-checks/src/test/java/org/sonar/javascript/checks/NoNestedFunctionsCheckTest.java:1]: Previously, there was no test at all for this one.

File: sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/GetterSetterCheck.java [hunk 1]
```diff
@@ -43,11 +43,22 @@ public class GetterSetterCheck implements EslintBasedCheck {
 
   @Override
   public List<Object> configurations() {
-    return Collections.singletonList(getWithoutSet);
+    return Collections.singletonList(
+      new Config(getWithoutSet)
```
Comment (zglicz) [inline sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/GetterSetterCheck.java:47]: this.

File: sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/HardcodedCredentialsCheck.java [hunk 2]
```diff
@@ -43,11 +44,22 @@ public class HardcodedCredentialsCheck implements EslintBasedCheck {
 
   @Override
   public List<Object> configurations() {
-    return Arrays.asList((Object[]) credentialWords.split("\\s*,\\s*"));
+    return Collections.singletonList(
+      new Config(credentialWords.split("\\s*,\\s*"))
```
Comment (zglicz) [inline sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/HardcodedCredentialsCheck.java:48]: this or not this?

**Review by zglicz — APPROVED** [review]
Talked offline about the typing - very slick.
The Java class serve no purpose, except for storing the default values. Except for that, they could be autogenerated. If we were to move these definitions into js/ts, we could fully auto generate them and remove the need for the additional tests to test passing of simple values....

All in all looks good and a very positive chang
lgtmacro
e!

File: packages/jsts/src/rules/S107/rule.ts [hunk 3]
```diff
@@ -45,7 +62,7 @@ export const rule: Rule.RuleModule = {
     const ruleDecoration: Rule.RuleModule = interceptReport(
       eslintMaxParams,
       function (context: Rule.RuleContext, descriptor: Rule.ReportDescriptor) {
-        const maxParams = context.options[0] as number;
+        const maxParams = (context.options as Options)[0].maximumFunctionParameters;
```
[the remark this answers is not in the document]
Comment (ericmorand-sonarsource) [reply packages/jsts/src/rules/S107/rule.ts:65]: Good catch, I'll use destructuring.

File: packages/jsts/src/rules/S2004/rule.ts [hunk 4]
```diff
@@ -25,21 +25,36 @@ import { TSESTree } from '@typescript-eslint/utils';
 import { getMainFunctionTokenLocation } from 'eslint-plugin-sonarjs/lib/utils/locations';
 import { SONAR_RUNTIME } from '../../linter/parameters';
 import { RuleContext, toEncodedMessage } from '../helpers';
+import type { RuleModule } from '../../../../shared/src/types/rule';
 
 const DEFAULT_THRESHOLD = 4;
```
[the remark this answers is not in the document]
Comment (ericmorand-sonarsource) [reply packages/jsts/src/rules/S2004/rule.ts:30]: I definitely agree.

File: sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/HardcodedCredentialsCheck.java (hunk 2 above)
Comment (ericmorand-sonarsource) [reply sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/HardcodedCredentialsCheck.java:48]: this :smile:

**Pull request merged**